docs/modules/ROOT/pages/firebase.adoc

```diff
@@ -0,0 +1,98 @@
+* EventArc
+
+Currently you can specify a custom `firebase.json` file but suport for this is limited. A
+future version will support reading the configuration from the `firebase.json` file
+instead of from the Quarkus configuration.
+
+== Bootstrapping the project
+
+First, we need a new project. Create a new project with the following command (replace the version placeholder with the correct one):
+
+[source, shell script]
+----
+mvn io.quarkus:quarkus-maven-plugin:${quarkusVersion}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=firebase-admin-quickstart \
+    -Dextensions="quarkus-google-cloud-firebase"
+cd firebase-admin-quickstart
+----
+
+This command generates a Maven project, importing the Google Cloud Firebase extension.
+
+If you already have your Quarkus project configured, you can add the `quarkus-google-cloud-firebase` extension to your project by running the following command in your project base directory:
+
+[source, shell script]
+----
+./mvnw quarkus:add-extension -Dextensions="quarkus-google-cloud-firebase"
+----
+
+This will add the following to your pom.xml:
+
+[source, xml]
+----
+<dependency>
+    <groupId>io.quarkiverse.googlecloudservices</groupId>
+    <artifactId>quarkus-google-cloud-firebase</artifactId>
+</dependency>
+----
+
+== Custom Docker image
+
+To run the emulators, a custom Docker image is build on the fly to run the Firebase emulators. This image is based on a NodeJS based image (refer to the configuration of the default value of `quarkus.google.cloud.firebase.devservice.image-name` to see the base image).
+
+You can configure a custom image if needed as base image to run the Firebase Emulators in. This image has the following requirements:
+
+* The image must support NodeJS in a version compatible with the required Firebase Tools
+* The image must be `alpine` based (or at least able to install the following packages using `apk`: )
+** openjdk11-jre
+** bash
+** curl
+** openssl
+** gettext
+** nano
+** nginx
+** sudo
+
+== Custom Firebase JSON
+
+By default a `firebase.json` file is generated inside the image to configure the various emulators. You can configure the DevService to use your own custom firebase.json file (e.g generated using the Firebase tools CLI). The following requirements are defined for this file:
+
+* Each of the emulators must be exposed on `0.0.0.0` as host as described https://firebase.google.com/docs/emulator-suite/use_hosting#emulators-no-local-host[here]. If this is not done, the Emulators will not be reachable from the Docker host.
+* Emulators need to be configured to use the default ports. Customizing the ports on which they run is currently not supported (this might change in a future version).
+
+== Interaction with other other extensions
+
+The following extensions support a DevService which conflicts with the DevService exposed by the Firebase Emulators.
+
+* Firestore
+* PubSub
+* TODO: Verify Storage
```

Comment:
Better to verify it as TODO in doc is not a good idea.

Reply:
Will look into that!

Generic question: in the mean time, I've been doing some work to create a separate module for the firebase-testcontainers stuff (i.e. move the definition of the Container to a separate module). It lives currently at https://github.com/alfa1-group/testcontainers-firebase 

I could change the current PR to use that module as dependency (reducing the size of the PR greatly, the majority of the work is in that container actually). 

Any pref on that?

Reply:
This is not an east to answer question ;) 

Usually, using an existing dependency is geat, but the bar to include a dependency in Quarkus is high so we want each dependency to have clear maintenance and stewardship, regular updates, security quality, ...

So a brand new repo from a not well known source didn't meet the bar for inclusion in Quarkus as a dependency, I hop you understand that I'm not judging the quality of the repo, the maintainer or the company behind it.

If you want to reuse the container, it may be possible to release it as a separate module with no dependencies so it could be reused outside this library.

Reply:
All good and fully understandable. What I'll do I copy the code from the repo to the PR and keep them in sync for now. 

Once the other repo gains enough credibility we can always choose to switch to that (or not and just keep the copied code as is, either is fine).

The other repo has already proven valuable anyhow as it allowed me to start using the code in a real life project and iron out some missing features (and a few small bugs), as well as improve on the api of the module itself. 

I'll finish up on a few of those changes, then copy the altered code back so we can finish this PR.

Reply:
Thanks!